src/contracts/middleware/StakeRegistry.sol

```diff
@@ -0,0 +1,481 @@
+    function setMinimumStakeForQuorum(uint8 quorumNumber, uint96 minimumStake) external onlyServiceManagerOwner {
+        minimumStakeForQuorum[quorumNumber] = minimumStake;
+    }
+
+    /**
+     * @notice Registers the `operator` with `operatorId` for the specified `quorumNumbers`.
+     * @param operator The address of the operator to register.
+     * @param operatorId The id of the operator to register.
+     * @param quorumNumbers The quorum numbers the operator is registering for, where each byte is an 8 bit integer quorumNumber.
+     * @dev access restricted to the RegistryCoordinator
+     * @dev Preconditions (these are assumed, not validated in this contract):
+     *         1) `quorumNumbers` has no duplicates
+     *         2) `quorumNumbers.length` != 0
+     *         3) `quorumNumbers` is ordered in ascending order
+     *         4) the operator is not already registered
+     */
+    function registerOperator(address operator, bytes32 operatorId, bytes calldata quorumNumbers) external virtual {
+        _registerOperator(operator, operatorId, quorumNumbers);
+    }
+
+    /**
+     * @notice Deregisters the operator with `operatorId` for the specified `quorumNumbers`.
+     * @param operatorId The id of the operator to deregister.
+     * @param quorumNumbers The quourm numbers the operator is deregistering from, where each byte is an 8 bit integer quorumNumber.
+     * @dev access restricted to the RegistryCoordinator
+     * @dev Preconditions (these are assumed, not validated in this contract):
+     *         1) `quorumNumbers` has no duplicates
+     *         2) `quorumNumbers.length` != 0
+     *         3) `quorumNumbers` is ordered in ascending order
+     *         4) the operator is not already deregistered
+     *         5) `quorumNumbers` is the same as the parameter use when registering
+     */
+    function deregisterOperator(bytes32 operatorId, bytes calldata quorumNumbers) external virtual {
+        _deregisterOperator(operatorId, quorumNumbers);
+    }
+
+    /**
+     * @notice Used for updating information on deposits of nodes.
+     * @param operators are the addresses of the operators whose stake information is getting updated
+     * @param operatorIds are the ids of the operators whose stake information is getting updated
+     * @param quorumBitmaps are the bitmap of the quorums that each operator in `operators` is part of
+     * @param prevElements are the elements before this middleware in the operator's linked list within the slasher
+     * @dev Precondition:
+     *          1) `quorumBitmaps[i]` should be the bitmap that represents the quorums that `operators[i]` registered for
+     * @dev reverts if there are no operators registered with index out of bounds
+     */
+    function updateStakes(address[] calldata operators, bytes32[] calldata operatorIds, uint192[] calldata quorumBitmaps, uint256[] calldata prevElements) external {
```

Comment:
possible bug: looks like there isn't any check performed on the `quorumBitmaps` input.
(also seems a little odd that it's a uint192[] input type instead of a uint256[] ?)

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/fc499a5a79d335dd65d9f9add3e00d4efdc94ec5 i think this was a bug. you could add to the total stake without being slashable because its not in ur quorum bitmap